lib/_stream_writable.js

```diff
@@ -294,34 +294,41 @@ function validChunk(stream, state, chunk, cb) {
 
 Writable.prototype.write = function(chunk, encoding, cb) {
   const state = this._writableState;
-  var ret = false;
-  const isBuf = !state.objectMode && Stream._isUint8Array(chunk);
-
-  // Do not use Object.getPrototypeOf as it is slower since V8 7.3.
-  if (isBuf && !(chunk instanceof Buffer)) {
-    chunk = Stream._uint8ArrayToBuffer(chunk);
-  }
 
   if (typeof encoding === 'function') {
     cb = encoding;
     encoding = null;
   }
 
-  if (isBuf)
-    encoding = 'buffer';
-  else if (!encoding)
+  if (!encoding) {
```

Comment:
This could be combined with the if above by using an `else` and setting the `encoding` to `state.defaultEncoding` instead of `null`.